packages/backend/src/core/UptimeUpdater.ts

```diff
@@ -0,0 +1,95 @@
+import { Logger, TaskQueue } from '@l2beat/common'
+import { ProjectUptime } from '@l2beat/config'
+import { EthereumAddress } from '@l2beat/types'
+
+import { ProjectInfo } from '../model'
+import { ApiMonitor } from '../peripherals/uptime/ApiMonitor'
+import { RpcMonitor } from '../peripherals/uptime/RpcMonitor'
+import { Clock } from './Clock'
+
+export class UptimeUpdater {
+  private taskQueue = new TaskQueue<void>(() => this.update(), this.logger)
+
+  constructor(
+    private rpcMonitor: RpcMonitor,
+    private apiMonitor: ApiMonitor,
+    private clock: Clock,
+    private projects: ProjectInfo[],
+    private logger: Logger,
+  ) {
+    this.logger = this.logger.for(this)
+  }
+
+  start() {
+    this.logger.info('Started')
+    return this.clock.onNewMinute(() => {
+      this.taskQueue.addToFront()
```

Comment:
Maybe only add if there isn't anything running? We don't want the tasks to pile up if something stalls